include/dlaf/communication/kernels/all_reduce.h

```diff
@@ -14,65 +14,74 @@
+#include <pika/mpi.hpp>
 #include <pika/unwrap.hpp>
 
 #include "dlaf/common/callable_object.h"
 #include "dlaf/common/contiguous_buffer_holder.h"
 #include "dlaf/common/pipeline.h"
 #include "dlaf/communication/communicator.h"
-#include "dlaf/communication/executor.h"
 #include "dlaf/communication/message.h"
 #include "dlaf/communication/rdma.h"
-#include "dlaf/executors.h"
 #include "dlaf/matrix/tile.h"
+#include "dlaf/sender/transform_mpi.h"
 
 namespace dlaf {
 namespace comm {
 
 namespace internal {
 
 template <class T>
-auto allReduce(const common::PromiseGuard<comm::Communicator>& pcomm, MPI_Op reduce_op,
-               common::internal::ContiguousBufferHolder<const T> cont_buf_in,
-               common::internal::ContiguousBufferHolder<T> cont_buf_out,
-               const matrix::Tile<const T, Device::CPU>&, MPI_Request* req) {
+auto allReduce(common::PromiseGuard<comm::Communicator> pcomm, MPI_Op reduce_op,
+               common::internal::ContiguousBufferHolder<const T>& cont_buf_in,
+               common::internal::ContiguousBufferHolder<T>& cont_buf_out, MPI_Request* req) {
   auto& comm = pcomm.ref();
   auto msg_in = comm::make_message(cont_buf_in.descriptor);
   auto msg_out = comm::make_message(cont_buf_out.descriptor);
 
   DLAF_MPI_CHECK_ERROR(MPI_Iallreduce(msg_in.data(), msg_out.data(), msg_in.count(), msg_in.mpi_type(),
                                       reduce_op, comm, req));
-  return cont_buf_out;
+
+  return std::move(cont_buf_out);
 }
 
 DLAF_MAKE_CALLABLE_OBJECT(allReduce);
 
 template <class T>
-auto allReduceInPlace(const common::PromiseGuard<comm::Communicator>& pcomm, MPI_Op reduce_op,
-                      common::internal::ContiguousBufferHolder<T> cont_buf, MPI_Request* req) {
+auto allReduceInPlace(common::PromiseGuard<comm::Communicator> pcomm, MPI_Op reduce_op,
+                      common::internal::ContiguousBufferHolder<T>& cont_buf, MPI_Request* req) {
   auto& comm = pcomm.ref();
   auto msg = comm::make_message(cont_buf.descriptor);
 
   DLAF_MPI_CHECK_ERROR(
       MPI_Iallreduce(MPI_IN_PLACE, msg.data(), msg.count(), msg.mpi_type(), reduce_op, comm, req));
-  return cont_buf;
+
+  return std::move(cont_buf);
 }
 
 DLAF_MAKE_CALLABLE_OBJECT(allReduceInPlace);
 }
 
 template <class T>
-void scheduleAllReduce(const comm::Executor& ex,
-                       pika::future<common::PromiseGuard<comm::Communicator>> pcomm, MPI_Op reduce_op,
+void scheduleAllReduce(pika::future<common::PromiseGuard<comm::Communicator>> pcomm, MPI_Op reduce_op,
                        pika::shared_future<matrix::Tile<const T, Device::CPU>> tile_in,
                        pika::future<matrix::Tile<T, Device::CPU>> tile_out) {
-  using pika::dataflow;
+  using pika::execution::experimental::keep_future;
+  using pika::execution::experimental::let_value;
+  using pika::execution::experimental::start_detached;
+  using pika::execution::experimental::transfer;
+  using pika::execution::experimental::when_all;
```

Comment:
```suggestion
  namespace ex = pika::execution::experimental;
```
?

Reply:
I replaced this and a few other places of `using namespace pika::execution::experimental` with `namespace ex = pika::execution::experimental`.